data-prepper-plugins/kinesis-source/src/main/java/org/opensearch/dataprepper/plugins/kinesis/source/configuration/KinesisStreamConfig.java

```diff
@@ -0,0 +1,32 @@
+package org.opensearch.dataprepper.plugins.kinesis.source.configuration;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotNull;
+import lombok.Getter;
+import software.amazon.kinesis.common.InitialPositionInStream;
+
+@Getter
+public class KinesisStreamConfig {
+    // Checkpointing interval
+    private static final int MINIMAL_CHECKPOINT_INTERVAL_MILLIS = 2 * 60 * 1000; // 2 minute
+    private static final boolean DEFAULT_ENABLE_CHECKPOINT = false;
+
+    @JsonProperty("stream_name")
+    @NotNull
+    @Valid
+    private String name;
+
+    @JsonProperty("stream_arn")
+    private String arn;
+
+    @JsonProperty("initial_position")
+    private InitialPositionInStream initialPosition = InitialPositionInStream.LATEST;
```

Comment:
Let's support the convention of lowercase and hyphens here as well. You can either:

1. Make a JSON mapping for the enums; or
2. Create your own enum that takes our name and uses the Kinesis enum.

e.g.

```
enum InitialPositionInStreamConfig {
  LATEST("latest", InitialPositionInStream.LATEST),
  ...
```

Reply:
Thanks @dlvenable. I have fixed this.